openmc/mgxs/library.py

```diff
@@ -717,3 +725,307 @@ def load_from_file(filename='mgxs', directory='mgxs'):
+            resultant mgxs data file. Defaults to "set1", "set2", ...
+        xs_ids : str or Iterable of str
+            Cross section set identifier (i.e., "71c") for all
+            data sets (if only str) or for each individual one
+            (if iterable of str). Defaults to '1g'
+        filename : str
+            Filename for the pickle file. Defaults to 'mg_cross_sections'.
+        directory : str
+            Directory for the pickle file. Defaults to './' (the
+            current working directory).
+        return_names : bool
+            Flag to indicate if the user would like the names of the
+            materials generated by this function returned with completion.
+            Defaults to True.
+
+        Returns
+        -------
+        mat_names : Iterable of str
+            Iterable of material names generated during this routine and
+            applies to the cross section library.  Note this is returned if
+            the return_names parameter is provided.
+
+        Raises
+        ------
+        ValueError
+            When the Library object is initialized with insufficient types of
+            cross sections for the Library.
+
+        See also
+        --------
+        Library.dump_to_file(mgxs_lib, filename, directory)
+
+        """
+
+        # Check the provided parameters
+        cv.check_value('xs_type', xs_type, ['macro', 'micro'])
+        if domain_names is not None:
+            cv.check_iterable_type('domain_names', filename, basestring)
+        if xs_ids is not None:
+            if isinstance(xs_ids, basestring):
+                # If we only have a string lets convert it now to a list
+                # of strings.
+                xs_ids = [xs_ids for i in range(len(self.domains))]
+            else:
+                cv.check_iterable_type('xs_ids', xs_ids, basestring)
+        else:
+            xs_ids = ['1g' for i in range(len(self.domains))]
+        cv.check_type('filename', filename, basestring)
+        cv.check_type('directory', directory, basestring)
+
+        # Construct the collection of the nuclides to report
+        if self.by_nuclide:
+            nuclides = self.all_mgxs[1][self.mgxs_types[-1]].get_all_nuclides()
+        else:
+            xs_type = 'macro'
+
+        # Make directory if it does not exist and build our filename
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+        full_filename = os.path.join(directory, filename + '.xml')
+        full_filename = full_filename.replace(' ', '-')
+
+        # Initialize file
+        mgxs_file = openmc.MGXSLibrary(self.energy_groups)
+
+        # Set the scattering order as isotropic until
+        # support for higher orders are included in openmc.mgxs
+        order = 0
+
+        # Build XSdata objects
+        xsdatas = []
+
+        mat_names = {}
+        for i in range(len(self.domains)):
+
+            id = self.domains[i].id
```

Comment:
Use `domain_id` in place of `id` to avoid colliding with the built-in `id()` function in Python.